bin/desi_qso_mgii_afterburner

```diff
@@ -0,0 +1,250 @@
+                             " / all (All objects which are tested by the mgII fitter)" +
+                             " --> To have 500 objects in the ouput file: set --target_selection all_targets --save_target all")
+
+    parser.add_argument("--lambda_width", type=str, required=False, default=250,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--max_sigma", type=str, required=False, default=200,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--min_sigma", type=str, required=False, default=10,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--min_deltachi2", type=str, required=False, default=16,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--min_signifiance_A", type=str, required=False, default=3,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+    parser.add_argument("--min_A", type=str, required=False, default=0.0,
+                        help="parameter for mgII fitter, see mgii_afterburner.py for more information")
+
+    return parser.parse_args()
+
+
+def select_targets_with_mgii_fitter(zbest, fibermap, index_sorted, sel_to_mgii, spectra_name, zbest_name, param_mgii_fitter, DESI_TARGET, save_target):
+    """
+    Run QuasarNet to the object with index_to_QN == True from spectra_name.
+    Then, Re-Run RedRock for the targetids which are selected by QN as a QSO.
+    Args:
+        zbest: fitsio hdu 'ZBEST' from zbest file
+        fibermap:  fitsio hdu 'FIBERMAP' from zbest file
+        index_sorted (int array): index to have fibermap[index_sorted] in the same order than zbest
+                      it is build with the TARGETID increasing
+        sel_to_mgii (bool array): size 500. Select on which objects mgii will be apply (index based on zbest)
+        spectra_name /zbest_name (str): The name of the spectra / associated zbest file
+        param_mgii_fitter (dict): contains info for the MgII fitter as lambda_width, max_sigma
+                                  min_sigma, min_deltachi2, min_signifiance_A, min_A
+        DESI_TARGET (str): name of DESI_TARGET for the wanted version of the target selection
+        save_target (str) : restricted (save only IS_QSO_MGII==true targets) / all (save all the sample)
+    Returns:
+        QSO_sel (pandas dataframe): contains all the information useful to build the QSO cat
+    """
+
+    # just to speed up the process and don't go in mgii_fitter function for nothing..
+    if sel_to_mgii.sum() == 0:
+        QSO_sel = pd.DataFrame()
+    else:
+        (index_selected_with_mgii_fit,
+        fits_result,
+        index_with_mgii_fit) = mgii_fitter(spectra_name, zbest_name, sel_to_mgii,
```

Comment:
Since the mgii fit is semi-expensive, please check that the `save_targets` input is a valid option before running the fit instead of risking running a long fit only to crash later because `save_targets = 'everything'`

Reply:
I check the validity of `--save_target` before running the function `select_targets_with_mgii_fitter`. Add this also in the qn afterburner